functors_task_1/student/include/impl.h

```diff
@@ -1,13 +1,22 @@
 #pragma once
 #include "bar_serving.h"
 #include <functional>
+#include <set>
+#include <map>
 
 /**
  * @todo Implement functor-generator that will return next beer brand (cyclic)
  * @example HoeGaarden -> Corona -> Carlsberg -> Bud -> ZlataPraha -> Leffe -> HoeGaarden...
  */
 struct BeerOrganizer
 {
+private:
+    int m_currentIndex = 0;
+public:
+    BeerBrand operator()() {
+        m_currentIndex = m_currentIndex == static_cast<int>(BeerBrand::Leffe) ? 1 : ++m_currentIndex;
```

Comment:
1. Avoid magic constants
2. Specify the range with "to" and "from" variables
3. Use "special" enum entries
4. Simplify condition with if statement